Src/Plugins/Security/AwsCredentialsValidator.cs

```diff
@@ -0,0 +1,147 @@
+                                               ref message);
+        }
+
+        public static string IsValidDynamic(ref string fingerprint, ref string message)
+        {
+            return ValidatorBase.IsValidDynamic(Instance,
+                                                ref fingerprint,
+                                                ref message);
+        }
+
+        protected override string IsValidStaticHelper(ref string matchedPattern,
+                                                      ref Dictionary<string, string> groups,
+                                                      ref string failureLevel,
+                                                      ref string fingerprintText,
+                                                      ref string message)
+        {
+            string id = groups["id"];
+            string key = groups["key"];
+
+            fingerprintText = new Fingerprint
+            {
+                Id = id,
+                Key = key,
+            }.ToString();
+
+            return nameof(ValidationState.Unknown);
+        }
+
+        protected override string IsValidDynamicHelper(ref string fingerprintText,
+                                                       ref string message)
+        {
+            var fingerprint = new Fingerprint(fingerprintText);
+
+            string id = fingerprint.Id;
+            string key = fingerprint.Key;
+
+            try
+            {
+                var iamClient = new AmazonIdentityManagementServiceClient(id, key);
+
+                GetAccountAuthorizationDetailsRequest request;
+                GetAccountAuthorizationDetailsResponse response;
+
+                GetUserRequest userRequest = new GetUserRequest();
+                GetUserResponse userResponse = iamClient.GetUserAsync().GetAwaiter().GetResult();
+
+                request = new GetAccountAuthorizationDetailsRequest();
+                response = iamClient.GetAccountAuthorizationDetailsAsync(request).GetAwaiter().GetResult();
+
+                message = BuildAuthorizedMessage(id, response);
+            }
+            catch (AmazonIdentityManagementServiceException e)
+            {
+                switch (e.ErrorCode)
+                {
+                    case "InvalidClientTokenId":
+                    case "SignatureDoesNotMatch":
+                    {
+                        return nameof(ValidationState.NoMatch);
+                    }
+                }
+
+                message = $"An unexpected exception was caught attempting to authenticate AWS id '{id}': {e.Message}";
+                return nameof(ValidationState.Unknown);
+            }
+            catch (Exception e)
+            {
+                message = $"An unexpected exception was caught attempting to authentic AWS id '{id}': {e.Message}";
+                return nameof(ValidationState.Unknown);
+            }
+
+/*          var client = new HttpClient();
```

Comment:
We should be able to replace use of the AWS SDK with direct REST API calls, just didn't do sufficient research to make it happen. Relevant code is at [https://github.com/aws/aws-sdk-net/](https://github.com/aws/aws-sdk-net/), specifically, the code in t\sdk\src\Core\Amazon.Runtime\Internal\Auth\AWS4Signer.cs